cql3/type_json.cc

```diff
@@ -498,66 +498,69 @@ static sstring to_json_string_aux(const user_type_impl& t, bytes_view bv) {
 namespace {
 struct to_json_string_visitor {
     bytes_view bv;
-    sstring operator()(const reversed_type_impl& t) { return to_json_string(*t.underlying_type(), bv); }
-    template <typename T> sstring operator()(const integer_type_impl<T>& t) { return to_sstring(compose_value(t, bv)); }
-    template <typename T> sstring operator()(const floating_type_impl<T>& t) {
+    value_and_type operator()(const reversed_type_impl& t) { return to_json_value(*t.underlying_type(), bv); }
```

Comment:
This is a big patch, and I don't understand it at all... Isn't it possible not to change **anything** in the existing code, and do something like
```c++
value_and_type to_json_value(const abstract_type &t, const managed_bytes_view& bv) {
   return value_and_type(to_json_string(t, bv), to_json_type(t))
}
```
Where to_json_type() converts the CQL type to the rjson::type you want?

Reply:
I dropped these changes and added `to_json_type()` instead in v3.